### ISSUE TITLE
dde-kwin fails to build

### ISSUE THREAD

**Issue by SamLukeYes** [linuxdeepin/dde-nixos#5]
### Describe the bug

`dde-kwin` fails to build while trying to build the VM from this repository.

### Metadata

 - system: `"x86_64-linux"`
 - host os: `Linux 5.15.79, NixOS, 22.11 (Raccoon), 22.11.20221120.690ffff`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.11.0`
 - nixpkgs: `/nix/store/0p5dnqxcz5zwkn24mjaysddx6j3q269i-source`

### To Reproduce

```command
$ nix run 'github:linuxdeepin/dde-nixos?dir=vm' --no-write-lock-file
```

### Logs

```
warning: not writing modified lock file of flake 'github:linuxdeepin/dde-nixos?dir=vm':
• Added input 'dde-nixos':
    'path:/nix/store/m8541bjfcklavwis7gvchlalzlij6yn6-source?lastModified=0&narHash=sha256-7M+ojo%2fQF3N+3VsxD2P1WmhQcK42NnDAKxJLPt9Lupo=' (1970-01-01)
• Added input 'dde-nixos/flake-utils':
    'github:numtide/flake-utils/5aed5285a952e0b949eb3ba02c12fa4fcfef535f' (2022-11-02)
• Added input 'dde-nixos/nixpkgs':
    'github:NixOS/nixpkgs/5ca8e2e9e1fa5e66a749b39261ad6bd0e07bc87f' (2022-11-12)
error: file 'nar/1cjbjw6xc8wrv0lznm2qkyjq1a7w26m5w5hq4v1h58rizmisafr4.nar.xz' does not exist in binary cache 'https://mirrors.bfsu.edu.cn/nix-channels/store'
error: file 'nar/1j5x8vq3xv0kzpdb7ga58cwja73p6wlrv720adyp6s42mdlq1arm.nar.xz' does not exist in binary cache 'https://mirrors.bfsu.edu.cn/nix-channels/store'
error: builder for '/nix/store/fi7w1r63cv4cxw8d2zq0z21q79ivx75x-dde-kwin-5.5.11-deepin.drv' failed with exit code 2;
       last 10 log lines:
       >                  from /nix/store/k0v9yvxjv42jbvkfvw4zmi5wjzbqzpf7-qtbase-5.15.7-dev/include/QtCore/QDebug:1,
       >                  from /build/source/plugins/platforms/plugin/libkwinpreload.cpp:25:
       > /nix/store/k0v9yvxjv42jbvkfvw4zmi5wjzbqzpf7-qtbase-5.15.7-dev/include/QtCore/qsharedpointer_impl.h:562:8: note: declared here
       >   562 |     T *data() const noexcept { return internalData(); }
       >       |        ^~~~
       > [ 91%] Built target dde-kwin-wayland
       > make[1]: *** [CMakeFiles/Makefile2:436: plugins/kdecoration/CMakeFiles/deepin-chameleon.dir/all] Error 2
       > [ 93%] Linking CXX shared module libdde-kwin-xcb.so
       > [ 96%] Built target dde-kwin-xcb
       > make: *** [Makefile:136: all] Error 2
       For full logs, run 'nix log /nix/store/fi7w1r63cv4cxw8d2zq0z21q79ivx75x-dde-kwin-5.5.11-deepin.drv'.
error: 1 dependencies of derivation '/nix/store/q0jsi5h911ql23z1niws1d8pfdhmlv7r-dbus-1.drv' failed to build
error: 1 dependencies of derivation '/nix/store/5qw51nw2k74pnknlz6ac50jhfxsy0bzy-startdde-5.9.50.drv' failed to build
error: 1 dependencies of derivation '/nix/store/l83rmv2378v8zlvwz9iwszw01lcqzl9j-system-path.drv' failed to build
error: 1 dependencies of derivation '/nix/store/v9s81fwkn9qrqai84rz94906l663vp7n-nixos-system-nixos-22.11.20221112.5ca8e2e.drv' failed to build
error: 1 dependencies of derivation '/nix/store/pd7mg81yk4yd5lpb2glsswpzchh9m3il-nixos-vm.drv' failed to build
```

The output of `nix log /nix/store/fi7w1r63cv4cxw8d2zq0z21q79ivx75x-dde-kwin-5.5.11-deepin.drv`:
- [dde-kwin.log](https://github.com/linuxdeepin/dde-nixos/files/10054504/dde-kwin.log)

**Comment by wineee** [comment]
related: https://github.com/linuxdeepin/developer-center/issues/3246

dde-kwin is really hard to port :( ,  [justforlxz](https://github.com/justforlxz) fix build for kwin 5.25 （But still can't run the  perfectly). This patch break since kwin upgrade to  5.26

justforlxz is working on deepin-kwin port to Debian/Arch， After completing this,  we don't need to worry about kde upgrade destroying dde-kwin.  This should be done next month

At present，you can revert https://github.com/linuxdeepin/dde-nixos/commit/f7e6624d665f766a1e449f0342a3b390dcca9039 to use kwin 5.25, or just use kde kwin as window manger

> by the way，Welcome to join the telegram group： https://t.me/ddeport

**Comment by SamLukeYes** [comment]
It works now, thanks!
![图片](https://user-images.githubusercontent.com/12882091/204542603-a9556b7b-c21a-44f4-95ba-3861d8461a40.png)
